src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -8874,4 +8939,100 @@ void GatherComponentLoadsSurfAbsFact(EnergyPlusData &state)
+        }
+        if (GndSurfaceTempSum == 0.0) {
+            GndSurfsProperty.SurfsTempAvg = 0.0;
+            state.dataSurface->UseSurfPropertyGndSurfTemp(SurfNum) = false;
+            continue;
+        }
+        GndSurfsProperty.SurfsTempAvg = root_4(GndSurfaceTempSum / GndSurfsProperty.SurfsViewFactorSum) - DataGlobalConstants::KelvinConv;
+    }
+}
+
+void GetGroundSurfacesReflectanceAverage(EnergyPlusData &state)
+{
+    //  returns ground surfaces average reflectance (dimenssionless)
+    //  ground reflectance viewed by a building exterior surface
+    //  ground surfaces reflectance weighed using view factors
+
+    // local vars
+    Real64 GndSurfRefl;
+    Real64 GndSurfsReflSum;
+
+    if (!state.dataGlobal->AnyLocalEnvironmentsInModel) {
+        return;
+    }
+    for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
+
+        if (!state.dataSurface->IsSurfPropertyGndSurfacesDefined(SurfNum)) continue;
+        auto &GndSurfsProperty = state.dataSurface->GroundSurfsProperty(state.dataSurface->GroundSurfsPropertyNum(SurfNum));
+        if (GndSurfsProperty.SurfsViewFactorSum == 0.0) {
+            state.dataSurface->UseSurfPropertyGndSurfRefl(SurfNum) = false;
+            continue;
+        }
+        GndSurfRefl = 0.0;
+        GndSurfsReflSum = 0.0;
+        for (int gSurfNum = 1; gSurfNum <= GndSurfsProperty.NumGndSurfs; gSurfNum++) {
+            if (GndSurfsProperty.GndSurfs(gSurfNum).ReflSchPtr == 0) continue;
+            GndSurfRefl = ScheduleManager::GetCurrentScheduleValue(state, GndSurfsProperty.GndSurfs(gSurfNum).ReflSchPtr);
+            GndSurfsReflSum += GndSurfsProperty.GndSurfs(gSurfNum).ViewFactor * GndSurfRefl;
+        }
+        if (GndSurfsReflSum == 0.0) {
+            GndSurfsProperty.SurfsReflAvg = 0.0;
+            state.dataSurface->UseSurfPropertyGndSurfRefl(SurfNum) = false;
+            continue;
+        }
+        GndSurfsProperty.SurfsReflAvg = GndSurfsReflSum / GndSurfsProperty.SurfsViewFactorSum;
+    }
+}
+
+void ReSetGroundSurfacesViewFactor(EnergyPlusData &state, int const SurfNum)
+{
+    //  resets ground veiew factors based on view factors indetity
+    //  the ground view factor value is set to the first element
+    //  when the ground view factor input field is blank
+
+    if (!state.dataSurface->IsSurfPropertyGndSurfacesDefined(SurfNum)) return;
+    auto &GndSurfsProperty = state.dataSurface->GroundSurfsProperty(state.dataSurface->GroundSurfsPropertyNum(SurfNum));
+    GndSurfsProperty.SurfsViewFactorSum = state.dataSurface->Surface(SurfNum).ViewFactorGroundIR;
+
+    if (GndSurfsProperty.SurfsViewFactorSum == 0.0) {
+        state.dataSurface->UseSurfPropertyGndSurfRefl(SurfNum) = false;
+        state.dataSurface->UseSurfPropertyGndSurfTemp(SurfNum) = false;
+        return;
+    }
+    GndSurfsProperty.GndSurfs(1).ViewFactor = GndSurfsProperty.SurfsViewFactorSum;
+}
```

Comment:
Function that resets the ground view factors if the ground view factor fields are blank.